### PR TITLE
Production ready logger module.

### PR DESCRIPTION
**Pull request by wtrocki**
## Motivation

Fix issues with logger module. Improve docs and tests.

## Progress
- [x] Remove invalid classes
- [x] Cleanup example
- [x] Fix format for bunyan logger
- [x] Rename/add clientLogger implementation
- [x] Fix problem with client side logger
- [x] Split interface and class (I was wrong with this) 
- [x] Typos and fixes for documentation
- [x] Fix unit tests and add assertions
- [x] Replace getter

### CODE REVIEW

**Review by JameelB — DISMISSED** [review]

File: cloud/logger/src/index.ts [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
+export { BunyanLogger } from './BunyanLogger';
+export { ClientLogger } from './ClientLogger';
+
+let logger: Logger = new EmptyLogger();
+
+function setLogger(userLogger: Logger) {
+  logger = userLogger;
+}
+
+export { logger, setLogger };
```
Comment (paolobueno) [inline cloud/logger/src/index.ts]: I'm not sure accessing `setLogger` would update the exported reference, might need a `getLogger()` instead.

Comment (paolobueno) [reply cloud/logger/src/index.ts]: i.e.
```typescript
import {logger, setLogger};
const newLogger = new BunyanLogger(...);
setLogger(newLogger);
newLogger !== logger; // ?
```

File: cloud/logger/README.md [hunk 2]
```diff
@@ -1,28 +1,34 @@
...
-
   logger.info('This log will render with BunyanLogger');
 
 ```
 
-See `./example/index.ts` for usage
+See `./example/index.ts` for more advanced usages
+
+
+
```
Comment (paolobueno) [inline cloud/logger/README.md]: too much whitespace :joy:

File: cloud/logger/src/index.ts (hunk 1 above)
Comment (wtrocki) [reply cloud/logger/src/index.ts]: I is updating it actually, but I think that it will be better to have getter and setter :)

Comment (paolobueno) [reply cloud/logger/src/index.ts]: :+1: yeah, thought it could work because it's essentially `exports.logger = newLogger` but it does get a bit weird

**Review by paolobueno — DISMISSED** [review]

**Review by austincunningham — APPROVED** [review]

**Review by JameelB — APPROVED** [review]

**Comment by wtrocki** [comment]
Verified in cloud and browser.

screen shot 2017-08-03 at 5 15 14 pm

**Pull request merged**